src/intrinsics.cr

```diff
@@ -156,12 +156,12 @@ module Intrinsics
     LibIntrinsics.counttrailing128({{src}}, {{zero_is_undef}})
   end
 
-  def self.va_start(ap)
-    LibIntrinsics.va_start(ap)
+  macro va_start(ap)
```

Comment:
Why is this change needed? When LibIntrinsic and module Intrinsic got split did that broke `VaList`?

Reply:
Using a function for va_start generates an error:

```
Invalid memory access (signal 11) at address 0x18
[0x558b0e4b6b36] *CallStack::print_backtrace:Int32 +118                                       
[0x558b0e49e831] __crystal_sigfault_handler +209                                              
[0x558b0f4c0cea] sigfault_handler +40                                                         
[0x7fd82d64b930] ???                                                                          
[0x7fd82e8e5733] _ZNK4llvm14MachineOperand5printERNS_11raw_ostreamERNS_17ModuleSlotTrackerENS_3LLTEbbbjPKNS_18TargetRegisterInfoEPKNS_19TargetIntrinsicInfoE +739
[0x7fd82e8c488c] _ZNK4llvm12MachineInstr5printERNS_11raw_ostreamERNS_17ModuleSlotTrackerEbbbbPKNS_15TargetInstrInfoE +1164
[0x7fd82e8c69e1] _ZNK4llvm12MachineInstr5printERNS_11raw_ostreamEbbbbPKNS_15TargetInstrInfoE +177
[0x7fd830c4dd08] ???                                                                          
[0x7fd830c4e2ae] ???                                                                          
[0x7fd82ed5cb31] _ZN4llvm10AsmPrinter16EmitFunctionBodyEv +561                                
[0x7fd830a66f8d] ???                                                                          
[0x7fd82e8bb535] _ZN4llvm19MachineFunctionPass13runOnFunctionERNS_8FunctionE +181
[0x7fd82e691530] _ZN4llvm13FPPassManager13runOnFunctionERNS_8FunctionE +1248
[0x7fd82e6915b2] _ZN4llvm13FPPassManager11runOnModuleERNS_6ModuleE +98                        
[0x7fd82e690836] _ZN4llvm6legacy15PassManagerImpl3runERNS_6ModuleE +870
[0x7fd82fd9c76f] ???                                                                          
[0x7fd82fd9c894] LLVMTargetMachineEmitToFile +244
[0x558b0f1d4ee9] *LLVM::TargetMachine#emit_to_file<LLVM::Module, String, LLVM::CodeGenFileType>:Bool +89
[0x558b0f1d4f2b] *LLVM::TargetMachine#emit_obj_to_file<LLVM::Module, String>:Bool +11
[0x558b0f385f37] *Crystal::Compiler::CompilationUnit#compile_to_object:(LLVM::Module | Nil) +887
[0x558b0f385a63] *Crystal::Compiler::CompilationUnit#compile:(LLVM::Module | Nil) +51
[0x558b0e49ad3d] ~procProc(Nil) +637
[0x558b0e5a00a1] *Fiber#run:(IO::FileDescriptor | Nil) +113
[0x558b0e4986b6] ~procProc(Fiber, (IO::FileDescriptor | Nil)) +6
[0x0] ???                                      
```

Reply:
@bcardiff these intrinsics implicitly relate to the surrounding function. Putting them in a different function won't work.

Reply:
I suspect that. I wasn't aware there was no test coverage for this module.